administration/src/components/generation/EakForm.tsx

```diff
@@ -32,44 +34,52 @@ const renderCardType: ItemRenderer<CardType> = (cardType, {handleClick, modifier
 };
 
 interface Props {
-    onRemove: () => void
+    model: CardCreationModel
+    onUpdate: (newModel: CardCreationModel | null) => void
 }
 
 const EakForm = (props: Props) => {
-    const values = Object.values(CardType);
-    const [selected, setSelected] = useState(values[0]);
-    const [expirationDate, setExpirationDate] = useState(new Date());
-
     return (
         <div>
             <Card>
                 <CardHeader>
-                    <Button minimal icon="cross" onClick={props.onRemove} />
+                    <Button minimal icon="cross" onClick={() => props.onUpdate(null)}/>
                 </CardHeader>
                 <FormGroup label="Vorname">
-                    <InputGroup placeholder="Vorname"/>
+                    <InputGroup placeholder="Vorname"
+                                autoFocus
+                                value={props.model.forename}
+                                onChange={(event: ChangeEvent<HTMLInputElement>) => props.onUpdate({
+                                    ...props.model,
+                                    forename: event.target.value
+                                })}/>
                 </FormGroup>
                 <FormGroup label="Nachname">
-                    <InputGroup placeholder="Nachname"/>
+                    <InputGroup placeholder="Nachname"
+                                value={props.model.surname}
+                                onChange={(event: ChangeEvent<HTMLInputElement>) => props.onUpdate({
+                                    ...props.model,
+                                    surname: event.target.value
+                                })}/>
                 </FormGroup>
                 <FormGroup label="Ablaufdatum">
-                    <DateInput placeholder="Ablaufdatum" value={expirationDate}
+                    <DateInput placeholder="Ablaufdatum"
+                               value={props.model.expirationDate}
                                parseDate={str => new Date(str)}
-                               onChange={value => setExpirationDate(value)}
+                               onChange={value => props.onUpdate({...props.model, expirationDate: value})}
                                formatDate={date => date.toLocaleDateString()}
+                               maxDate={add(Date.now(), {years: 99})}
+                               minDate={new Date()}
```

Comment:
I personally like the `Date.now()` better due to readability. Not worth putting effort in changing that however 😁

Reply:
Yeah true. Problem is JS: Date.now() returns a number (I guess unix timestamp) and new Date() returns a Date object :sweat_smile:

Reply:
Ah, okay. I guess we can not do much about that then 👍🏽😁